include/nuttx/net/icmp.h

```diff
@@ -113,8 +133,16 @@ struct icmp_hdr_s
 
   /* ICMP_ECHO_REQUEST and ICMP_ECHO_REPLY data */
 
-  uint16_t id;               /* Used to match requests with replies */
-  uint16_t seqno;            /* "  " "" "   " "      " "  " "     " */
+  union
+    {
+      struct
+        {
+          uint16_t id;      /* Used to match requests with replies */
+          uint16_t seqno;   /* "  " "" "   " "      " "  " "     " */
+        };
+
+      uint32_t data;
```

Comment:
This causes unaligned stores, because icmp is not guaranteed to be aligned to 4-bytes. None of the other parts of the icmp or ip headers use `unit32_t` anywhere, (rather they use `uint16_t[2]`).

Reply:
https://github.com/apache/incubator-nuttx/pull/5090 to fix.